metaflow_extensions/torchrun/plugins/__init__.py

```diff
@@ -9,30 +10,78 @@
 
 NODE_STARTED_VAR = "torchrun_node_started"
 
+class TorchRunExecutor:
+
+    def __init__(self, 
+                 pathspec,
+                 main_addr, 
+                 main_port, 
+                 num_nodes, 
+                 node_index, 
+                 ) -> None:
+        
+        self.torchrun_args = {
+            "rdzv-id": "123",
+            "rdzv_endpoint": "%s:%s" % (main_addr, main_port),
+            "nnodes": num_nodes, 
+            "master_addr": main_addr,
+            "master_port": main_port, # batch doesn't matter. k8s depends on port opened in @kubernetes jobset. 
+            "node_rank": node_index,
+            "rdzv-backend": "c10d",
+            "max-restarts": 3
+        }
+    
+    def run(self, entrypoint,  entrypoint_args=None, entrypoint_args_raw=None, nproc_per_node=1):
```

Comment:
Since we are already making these changes, we can do one additional thing. We can provide a dictionary argument that can help override the torch-run arguments. There are [many torch-run arguments](https://github.com/pytorch/pytorch/blob/194d9aa0f2ca9ee2c996af5c2839ed1dc3f06e46/torch/distributed/run.py#L392-L607), and we should allow users to customize that as they wish (if they want to, Otherwise we anyways manage)